Bottom: c0261de1bd63ed51ac4386bb4cf8cba6bf59a079
Top:    0206e41f7e1da4889c39c60ec6bf756a64d0406f
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-12 11:12:22 -0500

Add Gitlab Authentication


---
diff --git a/ansible-playbook/group_vars/all/credentials.yml b/ansible-playbook/group_vars/all/credentials.yml
index d476bf91..828dde00 100644
--- a/ansible-playbook/group_vars/all/credentials.yml
+++ b/ansible-playbook/group_vars/all/credentials.yml
@@ -5,6 +5,8 @@ buildsystem_database_build_password: '{{ lookup("password", playbook_dir + "/cre
 buildsystem_database_root_password:  '{{ lookup("password", playbook_dir + "/credentials/buildsystem/database/root"           , length=20) }}'
 buildsystem_ssh_prod_user_password:  '{{ lookup("password", playbook_dir + "/credentials/buildsystem/ssh/prod"                , length=20) }}'
 buildsystem_ssh_root_user_password:  '{{ lookup("password", playbook_dir + "/credentials/buildsystem/ssh/root"                , length=20) }}'
+gitlab_buildbot_auth_id:             '{{ lookup("password", playbook_dir + "/credentials/git/auth/buildbot_client_id") }}'
+gitlab_buildbot_auth_secret:         '{{ lookup("password", playbook_dir + "/credentials/git/auth/buildbot_secret") }}'
 gitlab_db_password:                  '{{ lookup("password", playbook_dir + "/credentials/git/db/user"                         , length=20) }}'
 gitlab_email_password:               '{{ lookup("password", playbook_dir + "/credentials/git/email/gitlab@novatech-llc.com"   , length=20) }}'
 gitlab_root_password:                '{{ lookup("password", playbook_dir + "/credentials/git/users/root"                      , length=20) }}'
diff --git a/docker/buildbot-master/Dockerfile b/docker/buildbot-master/Dockerfile
index 1baa3f49..a79778ef 100644
--- a/docker/buildbot-master/Dockerfile
+++ b/docker/buildbot-master/Dockerfile
@@ -18,7 +18,8 @@ RUN pip --no-cache-dir install \
     'buildbot-console-view' \
     'buildbot-grid-view' \
     'buildbot-waterfall-view' \
-    'buildbot-www'
+    'buildbot-www' \
+    'ldap3'
 
 # Create buildbot user
 ARG BUILDBOT_UID=1000
diff --git a/docker/buildbot-master/buildbot/master.cfg b/docker/buildbot-master/buildbot/master.cfg
index 0a3aed76..e5fc5149 100644
--- a/docker/buildbot-master/buildbot/master.cfg
+++ b/docker/buildbot-master/buildbot/master.cfg
@@ -49,6 +49,20 @@ mn = reporters.MailNotifier(
 c['services'].append(mn)
 
 ####### WEB SERVER
+userInfoProvider = util.LdapUserInfo(
+        uri=os.getenv('LDAP_URI'),
+        bindUser=os.getenv('LDAP_BIND_USER'),
+        bindPw=os.getenv('LDAP_BIND_PW'),
+        accountBase='ou=user,dc=novatech',
+        groupBase='ou=group,dc=novatech',
+        accountPattern='(&(objectClass=inetOrgPerson)(uid=%(username)s))',
+        accountFullName='displayName',
+        accountEmail='mail',
+        groupMemberPattern='(&(objectClass=groupOfNames)(member=%(dn)s))',
+        groupName='cn',
+        accountExtraFields=['employeeType'],
+)
+
 c['www'] = {
     'port': int(os.getenv('BUILDBOT_WEB_PORT',8010)),
     'plugins': {
@@ -56,13 +70,22 @@ c['www'] = {
         'console_view': True,
         'grid_view': True,
         },
-    'auth':util.UserPasswordAuth([("georgem","tgbyhn")]),
+    'auth': util.GitLabAuth(os.getenv('GITLAB_URI'),
+                            os.getenv('GITLAB_CLIENT_ID'),
+                            os.getenv('GITLAB_CLIENT_SECRET'),
+                            userInfoProvider=userInfoProvider),
     'authz':util.Authz(
         allowRules=[
             util.AnyControlEndpointMatcher(role="admins"),
             ],
         roleMatchers=[
-            util.RolesFromUsername(roles=['admins'], usernames=["georgem"]),
+            util.RolesFromUsername(
+                roles=['admins'],
+                usernames=[
+                    "george.mccollister",
+                    "joseph.lutz",
+                    "andrew.cooper",
+                ]),
             ],
         ),
     }
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
index c9623933..2b40e701 100644
--- a/roles/buildbot-master/defaults/main.yml
+++ b/roles/buildbot-master/defaults/main.yml
@@ -43,6 +43,12 @@ container_env:
   BUILDBOT_WORKER_NTEL_PASS: '{{ buildbot_worker_ntel_password }}'
   BUILDBOT_WORKER_PTXDIST: 'worker-ptxdist'
   BUILDBOT_WORKER_PTXDIST_PASS: '{{ buildbot_worker_ptxdist_password }}'
+  GITLAB_URI: 'https://{{ gitlab_hostname }}/'
+  GITLAB_CLIENT_ID: '{{ gitlab_buildbot_auth_id }}'
+  GITLAB_CLIENT_SECRET: '{{ gitlab_buildbot_auth_secret }}'
+  LDAP_URI: 'ldap://{{ openldap_hostname }}:389/'
+  LDAP_BIND_USER: 'cn=proxyagent,dc=novatech'
+  LDAP_BIND_PW: '{{ openldap_proxyagent_password }}'
 
 # the name of the container being started
 container_name: '{{ docker_name_prefix }}buildbot'
